### ISSUE TITLE
Real World Octree datasets bugs

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#1973]
## 🐛 Bug

This issue is to collect bug reports for using the `NAPARI_OCTREE` with real world datasets and will be updated as needed

**Comment by sofroniewn** [comment]
To start I get an error with the following

```python
import napari
import dask.array as da

path = 'https://s3.embassy.ebi.ac.uk/idr/zarr/v0.1/9822151.zarr'
resolutions = [da.from_zarr(path, component=str(i))[0, 0, 0] for i in list(range(11))]

with napari.gui_qt():
    viewer = napari.Viewer()
    viewer.add_image(resolutions)
```

```python-traceback
(base) czirwc1macos2701:image-demos nsofroniew$ python examples/covid19.py
Octree input data has 11 levels:
    Level 0: 167424 x 79360 = 13.3 billion pixels -> 654 x 310 = 202740 tiles
    Level 1: 83712 x 39680 = 3.3 billion pixels -> 327 x 155 = 50685 tiles
    Level 2: 41856 x 19840 = 830.4 million pixels -> 164 x 78 = 12792 tiles
    Level 3: 20928 x 9920 = 207.6 million pixels -> 82 x 39 = 3198 tiles
    Level 4: 10464 x 4960 = 51.9 million pixels -> 41 x 20 = 820 tiles
    Level 5: 5232 x 2480 = 13.0 million pixels -> 21 x 10 = 210 tiles
    Level 6: 2616 x 1240 = 3.2 million pixels -> 11 x 5 = 55 tiles
    Level 7: 1308 x 620 = 810960 pixels -> 6 x 3 = 18 tiles
    Level 8: 654 x 310 = 202740 pixels -> 3 x 2 = 6 tiles
    Level 9: 327 x 155 = 50685 pixels -> 2 x 1 = 2 tiles
    Level 10: 163 x 77 = 12551 pixels -> 1 x 1 = 1 tiles
tiles: 0 -> 4 create: 4 delete: 0 time: 698.876ms
WARNING: Error drawing visual <TiledImage at 0x7f8339adb550>
ERROR:root:Unhandled exception:
Traceback (most recent call last):
  File "/Users/nsofroniew/GitHub/vispy/vispy/app/backends/_qt.py", line 825, in paintGL
    self._vispy_canvas.events.draw(region=None)
  File "/Users/nsofroniew/GitHub/vispy/vispy/util/event.py", line 455, in __call__
    self._invoke_callback(cb, event)
  File "/Users/nsofroniew/GitHub/vispy/vispy/util/event.py", line 475, in _invoke_callback
    self, cb_event=(cb, event))
  File "/Users/nsofroniew/GitHub/vispy/vispy/util/event.py", line 471, in _invoke_callback
    cb(event)
  File "/Users/nsofroniew/GitHub/vispy/vispy/scene/canvas.py", line 217, in on_draw
    self._draw_scene()
  File "/Users/nsofroniew/GitHub/vispy/vispy/scene/canvas.py", line 266, in _draw_scene
    self.draw_visual(self.scene)
  File "/Users/nsofroniew/GitHub/vispy/vispy/scene/canvas.py", line 304, in draw_visual
    node.draw()
  File "/Users/nsofroniew/GitHub/vispy/vispy/scene/visuals.py", line 99, in draw
    self._visual_superclass.draw(self)
  File "/Users/nsofroniew/GitHub/vispy/vispy/visuals/visual.py", line 443, in draw
    self._vshare.index_buffer)
  File "/Users/nsofroniew/GitHub/vispy/vispy/visuals/shaders/program.py", line 101, in draw
    Program.draw(self, *args, **kwargs)
  File "/Users/nsofroniew/GitHub/vispy/vispy/gloo/program.py", line 533, in draw
    canvas.context.flush_commands()
  File "/Users/nsofroniew/GitHub/vispy/vispy/gloo/context.py", line 176, in flush_commands
    self.glir.flush(self.shared.parser)
  File "/Users/nsofroniew/GitHub/vispy/vispy/gloo/glir.py", line 572, in flush
    self._shared.flush(parser)
  File "/Users/nsofroniew/GitHub/vispy/vispy/gloo/glir.py", line 494, in flush
    parser.parse(self._filter(self.clear(), parser))
  File "/Users/nsofroniew/GitHub/vispy/vispy/gloo/glir.py", line 819, in parse
    self._parse(command)
  File "/Users/nsofroniew/GitHub/vispy/vispy/gloo/glir.py", line 787, in _parse
    ob.set_data(*args)
  File "/Users/nsofroniew/GitHub/vispy/vispy/gloo/glir.py", line 1537, in set_data
    raise ValueError("Type %r not allowed for texture" % data.dtype)
ValueError: Type dtype('>u2') not allowed for texture

Segmentation fault: 11
```

**Comment by sofroniewn** [comment]
I also get errors with the dataset created here https://github.com/napari/napari/issues/1942#issue-751059521

**Comment by sofroniewn** [comment]
Ideally all the dtypes from the following test suite should work with the new layer https://github.com/napari/napari/blob/master/napari/_tests/test_dtypes.py

**Comment by sofroniewn** [comment]
> Ideally all the dtypes from the following test suite should work with the new layer https://github.com/napari/napari/blob/master/napari/_tests/test_dtypes.py

Closed by #2362
